libs/blocks/commerce/commerce.js

```diff
@@ -0,0 +1,139 @@
+      resolve();
+    });
+});
+
+export const formatOfferDetails = (o) => {
+  const formattedOffer = {};
+
+  function formatPrice(price, format) {
+    const currency = format.match(/'([^']+)'/);
+    return `${currency[1]}${price}`;
+  }
+
+  const {
+    offerType,
+    offerId,
+    productArrangementCode,
+    pricePoint,
+    customerSegment,
+    commitment,
+    term,
+    offerSelectorIds,
+    priceDetails,
+  } = o;
+
+  formattedOffer.offerType = offerType;
+  formattedOffer.offerId = offerId;
+  formattedOffer.productArrangementCode = productArrangementCode;
+  formattedOffer.pricePoint = pricePoint;
+  formattedOffer.customerSegment = customerSegment;
+  formattedOffer.commitment = commitment;
+  formattedOffer.term = term;
+  formattedOffer.offerSelectorIds = offerSelectorIds;
+  formattedOffer.price = formatPrice(priceDetails.price, priceDetails.formatString);
+  return formattedOffer;
+};
+
+export const decorateOfferDetails = async (el, of, searchParams) => {
+  function formatOfferDetailKeys(str) {
+    const details = str.split(/(?=[A-Z])/);
+    const allCapsDetail = details.map((detail) => detail.toUpperCase());
+    const result = allCapsDetail.join(' ');
+    return result;
+  }
+  const offerDetailsList = document.createElement('ul');
+  offerDetailsList.className = 'offer-details';
+  const offer = formatOfferDetails(of);
+  Object.entries(offer).forEach(([key, value]) => {
+    const offerData = document.createElement('li');
+    const offerKey = document.createElement('span');
+    const offerValue = document.createElement('span');
+    offerData.classList.add('offer-detail');
+    offerKey.classList.add('offer-key');
+    offerValue.classList.add('offer-value');
+    offerKey.textContent = `${formatOfferDetailKeys(key)}: `;
+    offerValue.textContent = value;
+    offerData.appendChild(offerKey);
+    offerData.appendChild(offerValue);
+    offerDetailsList.appendChild(offerData);
+  });
+  const checkoutLink = document.createElement('a');
+  checkoutLink.textContent = 'Checkout link';
+  const perpetual = searchParams.get('perp') === 'true' || undefined;
+  const promotionCode = searchParams.get('promo') || undefined;
+  const options = omitNullValues({
+    perpetual,
+    promotionCode,
+    wcsOsi: searchParams.get('osi'),
+    ...getCheckoutContext(searchParams, getConfig()),
+  });
+  const checkoutUrl = buildCheckoutButton(checkoutLink, options);
+  checkoutUrl.target = '_blank';
+  offerDetailsList.appendChild(checkoutUrl);
+  el.append(offerDetailsList);
+};
+
+export const handleSearch = async (e, el) => {
+  let searchParams = {};
+  function displaySearchError() {
+    const notValidUrl = document.createElement('h4');
+    notValidUrl.classList.add('not-valid-url');
+    notValidUrl.textContent = 'Not a valid offer link';
+    el.append(notValidUrl);
+  }
+  el.textContent = '';
+  const search = e.target.value;
+  if (search === '') return undefined;
+  try {
```

Comment:
if you use `try` to catch an exception from an async function, I recommend to `await` it.
Otherwise,  just chain the promise with .catch function at the line 114.

`window.tacocat.wcs.resolveOfferSelector(osi).then(([offerDetails]) => {}).catch(displaySearchError)`

Reply:
Chained `.catch()` to promise ✅